internal/controller/api/connection_mediator_v2_test.go

```diff
@@ -0,0 +1,108 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/RedHatInsights/cloud-connector/internal/config"
+	"github.com/RedHatInsights/cloud-connector/internal/domain"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	MESSAGE_ENDPOINT_V2 = URL_BASE_PATH + "/v2/connections/345/message"
+	ACCOUNT_NUMBER      = domain.AccountID("1234")
+)
+
+func MockedGetConnectionByClientID(context.Context, *logrus.Entry, domain.OrgID, domain.ClientID) (domain.ConnectorClientState, error) {
```

Comment:
What do you think about modifying this so that we can pass in the expected org-id, account and client-id?  I think that would be a cleaner approach and would likely be useful moving forward.  One way to approach this would be the following:

```go
func mockedGetConnectionByClientID(expectedOrgId domain.OrgID, expectedAccount domain.AccountID, expectedClientId domain.ClientID) connection_repository.GetConnectionByClientID {
	return func(ctx context.Context, log *logrus.Entry, actualOrgId domain.OrgID, actualClientId domain.ClientID) (domain.ConnectorClientState, error) {
		if actualOrgId != expectedOrgId {
			return domain.ConnectorClientState{}, fmt.Errorf("Actual org id does not match expected org id")
		}

		if actualClientId != expectedClientId {
			return domain.ConnectorClientState{}, fmt.Errorf("Actual client id does not match expected client id")
		}

		return domain.ConnectorClientState{OrgID: actualOrgId, Account: expectedAccount, ClientID: actualClientId}, nil
	}
}
```

This allows us to pass in the `expected` values and confirm that those expected values match how the code is called during the test.

It also allows us to make the tests a bit more self-contained (less globals, less hard coded values, etc).

Also notice that I'm not exporting the mock out of the package (little 'm' in mockedGetConnectionByClientID).  Always try to expose a little as possible.

Reply:
Great idea! Updated!